File: NeetCode/17-Bit-Manipulation/ReverseBits.java
public class ReverseBits {
    public int reverseBits(int n) {
        int result = 0;
        for(int i = 0; i < 32; i++){
            int temp = n & 1;
            result = result << 1;
            result = result | temp;
            n = n >>> 1;
        }
        return result;
    }
}

File: NeetCode/17-Bit-Manipulation/ReverseBits.cpp
class Solution {
public:
    int reverseBits(int n) {
        unsigned int result = 0;
        unsigned int un = n;
        for(int i = 0; i < 32; i++){
            int temp = un & 1;
            result = result << 1;
            result = result | temp;
            un = un >> 1;
        }
        return result;
    }
    
};